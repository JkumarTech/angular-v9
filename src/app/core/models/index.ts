export * from './film.interface';
export * from './person.interface';
export * from './planet.interface';
