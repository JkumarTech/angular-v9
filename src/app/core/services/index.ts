export * from './film.service';
export * from './person.service';
export * from './planet.service';
export * from './window.service';
