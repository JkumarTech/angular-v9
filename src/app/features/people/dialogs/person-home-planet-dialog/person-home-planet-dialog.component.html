<h1 mat-dialog-title>{{ data.person.fields.name }}: Home Planet</h1>
<mat-dialog-content>
  <swr-planet [planet]="planet | async"></swr-planet>
</mat-dialog-content>
<mat-dialog-actions align="end">
  <button mat-button mat-dialog-close>Close</button>
</mat-dialog-actions>
