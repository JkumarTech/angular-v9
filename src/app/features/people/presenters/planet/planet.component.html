<h3>{{ planet?.fields.name }}</h3>
<ul>
  <li>Climate: {{ planet?.fields.climate }}</li>
  <li>Diameter: {{ planet?.fields.diameter }}</li>
  <li>Rotation Period: {{ planet?.fields.rotation_period }}</li>
  <li>Created: {{ planet?.fields.created }}</li>
  <li>Terrain: {{ planet?.fields.terrain }}</li>
  <li>Gravity: {{ planet?.fields.gravity }}</li>
  <li>Orbital Period: {{ planet?.fields.orbital_period }}</li>
  <li>Population: {{ planet?.fields.population }}</li>
</ul>
