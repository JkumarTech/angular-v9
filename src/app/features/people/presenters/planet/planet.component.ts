import { Component, Input } from '@angular/core';
import { Planet } from 'src/app/core';

@Component({
  selector: 'swr-planet',
  templateUrl: './planet.component.html',
  styleUrls: ['./planet.component.scss']
})
export class PlanetComponent {
  /** The planet to display. */
  @Input() planet: Planet;
}
