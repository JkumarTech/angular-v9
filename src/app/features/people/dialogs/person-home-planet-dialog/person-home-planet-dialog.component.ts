import {
  Component,
  Inject,
  ViewChild,
  ViewContainerRef,
  OnInit,
  OnDestroy,
  ComponentFactoryResolver,
  TemplateRef
} from '@angular/core';
import { MAT_DIALOG_DATA } from '@angular/material';
import { tap, takeUntil } from 'rxjs/operators';
import { Subject } from 'rxjs';

import { Person, PlanetService, Planet } from '../../../../core';

@Component({
  templateUrl: './person-home-planet-dialog.component.html',
  styleUrls: ['./person-home-planet-dialog.component.scss']
})
export class PersonHomePlanetDialogComponent implements OnDestroy, OnInit {
  /** The view container reference for the planet template. */
  @ViewChild(TemplateRef, { read: ViewContainerRef })
  planetViewContainerRef: ViewContainerRef;

  /** Unsubscribe from observable streams when the component is destroyed. */
  private unsubscribe = new Subject();

  constructor(
    private componentFactoryResolver: ComponentFactoryResolver,
    @Inject(MAT_DIALOG_DATA) public data: { person: Person },
    private readonly planetService: PlanetService
  ) {}

  ngOnDestroy() {
    this.unsubscribe.next();
    this.unsubscribe.complete();
  }

  ngOnInit() {
    this.planetService
      .getPlanetForPerson(this.data.person)
      .pipe(
        tap(planet => this.lazyLoadPlanet(planet)),
        takeUntil(this.unsubscribe)
      )
      .subscribe();
  }

  private lazyLoadPlanet(planet: Planet): void {
    import('../../presenters/planet/planet.component').then(
      ({ PlanetComponent }) => {
        const componentFactory = this.componentFactoryResolver.resolveComponentFactory(
          PlanetComponent
        );
        const componentRef = this.planetViewContainerRef.createComponent(
          componentFactory
        );
        componentRef.instance.planet = planet;
      }
    );
  }
}
