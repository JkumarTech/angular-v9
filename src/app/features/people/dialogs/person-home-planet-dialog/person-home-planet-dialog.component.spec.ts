import { CommonModule } from '@angular/common';
import { HttpClientTestingModule } from '@angular/common/http/testing';
import { ComponentFixture, async } from '@angular/core/testing';
import { MAT_DIALOG_DATA } from '@angular/material';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { RouterTestingModule } from '@angular/router/testing';
import { of } from 'rxjs';

import { ConfigureFn, configureTests } from '../../../../../test-config.helper';
import { Person, PlanetService } from '../../../../core';
import { MaterialModule } from '../../../../material';
import { SharedModule } from '../../../../shared';
import { PlanetComponent } from '../../presenters/planet/planet.component';
import { PersonHomePlanetDialogComponent } from './person-home-planet-dialog.component';

const person = {
  fields: {
    edited: '2014-12-20T21:17:56.891Z',
    name: 'Luke Skywalker',
    created: '2014-12-09T13:50:51.644Z',
    gender: 'male',
    skin_color: 'fair',
    hair_color: 'blond',
    height: '172',
    eye_color: 'blue',
    mass: '77',
    homeworld: 1,
    birth_year: '19BBY'
  },
  model: 'resources.people',
  pk: 1
};
const planet = {
  fields: {
    edited: '2014-12-20T20:58:18.411Z',
    climate: 'arid',
    surface_water: '1',
    name: 'Tatooine',
    diameter: '10465',
    rotation_period: '23',
    created: '2014-12-09T13:50:49.641Z',
    terrain: 'desert',
    gravity: '1 standard',
    orbital_period: '304',
    population: '200000'
  },
  model: 'resources.planet',
  pk: 1
};

describe('PersonHomePlanetDialogComponent', () => {
  let fixture: ComponentFixture<PersonHomePlanetDialogComponent>;
  let app: PersonHomePlanetDialogComponent;

  beforeEach(async(() => {
    const configure: ConfigureFn = testBed => {
      testBed.configureTestingModule({
        declarations: [PersonHomePlanetDialogComponent, PlanetComponent],
        imports: [
          CommonModule,
          HttpClientTestingModule,
          NoopAnimationsModule,
          RouterTestingModule,
          MaterialModule,
          SharedModule
        ],
        providers: [
          {
            provide: MAT_DIALOG_DATA,
            useValue: { person }
          },
          {
            provide: PlanetService,
            useValue: {
              getPlanetForPerson: jest.fn((_: Person) => of(planet))
            }
          }
        ]
      });
    };

    configureTests(configure).then(testBed => {
      fixture = testBed.createComponent(PersonHomePlanetDialogComponent);
      app = fixture.componentInstance;
      fixture.detectChanges();
    });
  }));

  it('should create the app', async(() => {
    expect(app).toBeTruthy();
  }));

  it('should render html', async(() => {
    const compiled = fixture.debugElement.nativeElement;
    // @ts-ignore
    expect(compiled).toMatchSnapshot();
  }));
});
