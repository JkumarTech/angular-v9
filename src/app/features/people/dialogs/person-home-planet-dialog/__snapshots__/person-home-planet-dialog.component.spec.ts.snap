// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonHomePlanetDialogComponent should render html 1`] = `
<div
  id="root1"
  ng-version="9.0.0-rc.6"
>
  <h1
    class="mat-dialog-title"
    id="mat-dialog-title-1"
    mat-dialog-title=""
  >
    Luke Skywalker: Home Planet
  </h1>
  <mat-dialog-content
    class="mat-dialog-content"
  >
    <swr-planet
      ng-reflect-planet="[object Object]"
    >
      <h3>
        Tatooine
      </h3>
      <ul>
        <li>
          Climate: arid
        </li>
        <li>
          Diameter: 10465
        </li>
        <li>
          Rotation Period: 23
        </li>
        <li>
          Created: Dec 9, 2014
        </li>
        <li>
          Terrain: desert
        </li>
        <li>
          Gravity: 1 standard
        </li>
        <li>
          Orbital Period: 304
        </li>
        <li>
          Population: 200000
        </li>
      </ul>
    </swr-planet>
  </mat-dialog-content>
  <mat-dialog-actions
    align="end"
    class="mat-dialog-actions"
  >
    <button
      class="mat-button mat-button-base _mat-animation-noopable"
      mat-button=""
      mat-dialog-close=""
      ng-reflect-dialog-result=""
      type="button"
    >
      <span
        class="mat-button-wrapper"
      >
        Close
      </span>
      <div
        class="mat-button-ripple mat-ripple"
        matripple=""
        ng-reflect-centered="false"
        ng-reflect-disabled="false"
        ng-reflect-trigger="[object HTMLButtonElement]"
      />
      <div
        class="mat-button-focus-overlay"
      />
    </button>
  </mat-dialog-actions>
</div>
`;
